src/snowcli/connection_config.py

```diff
@@ -28,3 +28,12 @@ def add_connection(self, connection_name, entry):
 
         with open(os.path.expanduser(self.path), "w+") as f:
             self.config.write(f)
+
+    def _replace_with_env_variables(self):
+        connection_sections = [
```

Comment:
WDYT about moving this filtering outside of `_replace_with_env_variables` function? IMO the function shouldn't have to worry about which configs it should operate on. IMO it should just do what the name states -> replace env variables for "configs" known in the context of the class. IMO we should do filtering before calling `_replace_with_env_variables`. The class in named `ConnectionConfigs` so we want to keep only connection configs and `_replace_with_env_variables` doesn't seem to me to be a good selection to be responsible for this. 

WDYT?

Reply:
You are right, but this class is responsible for adding new connections too. We have to keep whole config, because we write everything into file. For now I would extract filter part into separate function and leave the rest as is.

Reply:
A few notes from our meeting (I decided to write them down to not forget about anything):
* We have to change the implementation because currently after invocation of "add connection" we will save resolved env values to the file instead of keeping env variables keys.
* We can fix it by moving env vars resolution to "get connection".
* But probably even better solution would be to move it to new method "get connections". The method can be responsible for:
    * Config sections filtering
    * Doing all necessary replacements
* "get connections" can be invoked outside "ConnectionConfigs" class to get info about all possible connections (currently 1 usage) - this will allow to remove one of underlying sqlconfig leaks
* Old "get connection" method can invoke "get connections" and search only for the one connection with the given name. IMO it's worth to think about making "get connection" returning optional value (to allow to remove the second sqlconfig leak)
    * Alternative solution to remove this leak is to have additional method "containsConnection" or sth similar which will check if there is a connection with the name
* When we will have all of it, IMO it should be possible to try to make underlying sqlconfig a private field (I don't know how it will work with tests)
* Sometime in the future, when we switch from dictionary to dataclass for single connection config representation, I hope we won't have any leaks of underlying sqlconfig and we'll be able to simply switch from one underlying storage to another if there will be such need
* Not everything above has to be done in this PR. As far as we fix the issue from the first point, other points can be moved to separate task.

Reply:
+ It's worth to think about a test confirming that we fixed the issue mentioned above.